internal/pkg/upload/es.go

```diff
@@ -0,0 +1,172 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package upload
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/elastic/fleet-server/v7/internal/pkg/bulk"
+	"github.com/elastic/fleet-server/v7/internal/pkg/dsl"
+	"github.com/elastic/fleet-server/v7/internal/pkg/es"
+	"github.com/elastic/fleet-server/v7/internal/pkg/model"
+	"github.com/elastic/fleet-server/v7/internal/pkg/upload/cbor"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	// @todo: neither of these should be static. But should be specific to an integration
+	// somewhat configurable, but need to follow a pattern so that Fleet Server has write access
+	FileHeaderIndexPattern = ".fleet-%s-files"
+	FileDataIndexPattern   = ".fleet-%s-file-data"
```

Comment:
based on this conversation: https://github.com/elastic/security-team/issues/4661#issuecomment-1309075519

We should update these to be `.fleet-files-%s` and `.fleet-file-data-%s` .